src/platform_impl/windows/event_loop.rs

```diff
@@ -1961,6 +1961,106 @@ unsafe fn public_window_callback_inner<T: 'static>(
             commctrl::DefSubclassProc(window, msg, wparam, lparam)
         }
 
+        winuser::WM_NCCALCSIZE => {
+            let window_state = subclass_input.window_state.lock();
+
+            if !window_state.window_flags.contains(WindowFlags::DECORATIONS) {
+                // adjust the maximized borderless window fill the work area rectangle of the display monitor
+                if util::is_maximized(window) {
+                    let monitor = monitor::current_monitor(window);
+                    if let Ok(monitor_info) = monitor::get_monitor_info(monitor.hmonitor()) {
+                        let params = &mut *(lparam as *mut winuser::NCCALCSIZE_PARAMS);
+                        params.rgrc[0] = monitor_info.rcWork;
+                    }
+                }
+                0 // must return a value here, other wise the window won't be borderless (without decorations)
+            } else {
+                commctrl::DefSubclassProc(window, msg, wparam, lparam)
+            }
+        }
+
+        winuser::WM_NCHITTEST => {
```

Comment:
I'm not opposed to enabling resizing on borderless windows in principle, but there should be some thought as to how this is exposed to users of Winit, as well as how this might be implemented on other platforms. It's actually not clear to me whether the resizing area is on the window's "shadow", or if it intrudes on the client area.

See #725 and #221 for more info.

Reply:
It intrudes in the first few pixels in the client area. It is calculated by the cursor position relative to the window edge + the border size. we can even replace the border size with custom value or remove it.

I see some valid concerns about where the resize handler should appear.

And I don't know about other platforms tbh but I think macos does let you resize broderless windows.

Reply:
I really would prefer if custom resize areas were done in a separate PR, so that this one can be focused only on smoothing over a Windows quirk.

Reply:
Yeah, That makes sense, I can make this PR focus only on Fixing the aero-snap but it would still require either the workaround I purposed below or a fix for the subclass issue.